--- updateFirmwareFromFile.sh.orig	2011-09-03 03:40:45.000000000 +0400
+++ updateFirmwareFromFile.sh	2012-04-02 22:35:07.000000000 +0400
@@ -102,6 +102,36 @@
 old_color=`cat /usr/local/nas/led_color`
 echo white > /usr/local/nas/led_color
 #upgrade 
+
+OLD_PWD="$PWD"
+UPDATE_DIR="$(dirname ${filename})"
+UPDATE_NAME="$(basename ${filename})"
+PATCH_FILE="$(dirname $0)/wedroInfectUpdate.patch"
+CONTROL_ARCHIVE="$(ar t $UPDATE_DIR/$UPDATE_NAME | grep control)"
+mkdir -p /tmp/infection
+cd /tmp/infection
+ar x $UPDATE_DIR/$UPDATE_NAME $CONTROL_ARCHIVE
+mkdir control
+cd control
+tar -x -f ../$CONTROL_ARCHIVE
+patch -s -t -N postinst $PATCH_FILE > /dev/null 2>&1
+if [ -z "$(ls | grep -i '.rej')" ]; then
+    tar -c -f $CONTROL_ARCHIVE $(ls)
+    mv -f $CONTROL_ARCHIVE ../$CONTROL_ARCHIVE
+    REPACK='1'
+    echo '[QUO]: postinst script is patched. Rebuilding update package...'
+else
+    echo '[QUO]: something went wrong. Maybe script is already patched?'
+fi
+cd ..
+rm -rf control
+if [ "$REPACK" == '1' ]; then
+    ar r $UPDATE_DIR/$UPDATE_NAME $CONTROL_ARCHIVE
+fi
+rm $CONTROL_ARCHIVE
+cd $OLD_PWD
+rm -rf /tmp/infection
+
 dpkg -i ${filename} 2>&1 | tee ${updatelog} > /dev/null
 status=$?
 
