--- postinst	2011-10-19 05:32:19.000000000 +0400
+++ postinst.new	2012-03-08 23:35:25.271592080 +0300
@@ -73,6 +73,11 @@
 mount ${upgradeDevice} ${upgradeMountPath}
 sleep 2
 
+
+# QUO: "Infecting" installation with WD QUO Scripts
+wedroInfectRootfs.sh "${upgradeMountPath}"
+
+
 echo "Re-sourcing 'disk-param.sh' from (${upgradeMountPath}/usr/local/sbin/disk-param.sh) to aquire any changed or new definitions..."
 . ${upgradeMountPath}/usr/local/sbin/disk-param.sh
 
